contribs/vsp/src/main/java/playground/vsp/ev/UrbanVehicleChargingHandler.java

```diff
@@ -115,9 +115,14 @@ public void handleEvent(ActivityEndEvent event) {
 			if (tuple != null) {
 				Id<Vehicle> evId = Id.create(tuple.getFirst(), Vehicle.class);
 				if(vehiclesAtChargers.remove(evId) != null){ //if null, vehicle is fully charged and de-plugged already (see handleEvent(ChargingEndedEvent) )
+					//this is not necessarily true as vehicles can be also queued and waiting for plug in Nov23 Ashraf
 					Id<Charger> chargerId = tuple.getSecond();
 					Charger c = chargingInfrastructure.getChargers().get(chargerId);
 					c.getLogic().removeVehicle(electricFleet.getElectricVehicles().get(evId), event.getTime());
+				}else {
+					Id<Charger> chargerId = tuple.getSecond();
+					Charger c = chargingInfrastructure.getChargers().get(chargerId);
+					if(c.getLogic().getQueuedVehicles().contains(electricFleet.getElectricVehicles().get(evId)))c.getLogic().removeVehicle(electricFleet.getElectricVehicles().get(evId), event.getTime());// so it is important to remove the vehicle from queue as well.
```

Comment:
Please reformat and rearrange your code (e.g. by adding local variables) so it is easier to read.

Reply:
I have added a local variable for the EV for easier reading

Reply:
the code should be shorter and easier to read like this:

```
if(vehiclesAtChargers.remove(evId) != null || c.getLogic().getQueuedVehicles().contains(ev)){ 
	Id<Charger> chargerId = tuple.getSecond();
	Charger c = chargingInfrastructure.getChargers().get(chargerId);
	c.getLogic().removeVehicle(ev, event.getTime());
}
```